hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSNamesystem.java

```diff
@@ -9034,9 +9038,12 @@ private boolean isObserver() {
 
   private void checkBlockLocationsWhenObserver(LocatedBlocks blocks, String src)
       throws ObserverRetryOnActiveException {
-    for (LocatedBlock b : blocks.getLocatedBlocks()) {
-      if (b.getLocations() == null || b.getLocations().length == 0) {
-        throw new ObserverRetryOnActiveException("Zero blocklocations for " + src);
+    List<LocatedBlock> locatedBlockList = blocks.getLocatedBlocks();
```

Comment:
minor nit (non-blocking): can we move the null-check for `blocks` inside of this method? Then we only need it one place instead of duplicated three times

Reply:
OK, I have fix it.